plugins/core/Standard/qRANSAC_SD/qRANSAC_SD_Commands.h

```diff
@@ -0,0 +1,514 @@
+				QDir dir(arg);
+				if (!dir.exists())
+				{
+					cmd.print(QObject::tr("\n%1 Does not exist\tcreating path").arg(arg));
+					dir.mkpath(arg);
+				}
+				outputPairDir  = dir.cleanPath(arg);
+				cmd.print(QObject::tr("\t%1 : %2").arg(OUT_PAIR_DIR, outputPairDir));
+				outputIndividualPairs = true;
+				}
+				else if (param == OUTPUT_INDIVIDUAL_SUBCLOUDS)
+				{
+					outputIndividualClouds = true;
+				}
+				else if (param == OUTPUT_INDIVIDUAL_PRIMITIVES)
+				{
+					outputIndividualPrimitives = true;
+				}
+				else if (param == OUTPUT_INDIVIDUAL_PAIRED_CLOUD_PRIMITIVE)
+				{
+					outputIndividualPairs = true;
+				}
+				else if (param == OUTPUT_GROUPED)
+				{
+					outputGrouped = true;
+				}
+				else if (param == ENABLE_PRIMITIVE)
+				{
+					if (cmd.arguments().empty())
+					{
+						return cmd.error(QObject::tr("Missing parameter: primitive type after \"-%1 %2\"").arg(COMMAND_RANSAC, ENABLE_PRIMITIVE));
+					}
+					QString prim = cmd.arguments().takeFirst().toUpper();
+					unsigned primCount = 0;
+					while (primitiveNames.contains(prim))
+					{
+						if (prim == PRIM_PLANE)
+						{
+							params.primEnabled[qRansacSD::RPT_PLANE] = true;
+							primCount++;
+						}
+						else if (prim == PRIM_SPHERE)
+						{
+							params.primEnabled[qRansacSD::RPT_SPHERE] = true;
+							primCount++;
+						}
+						else if (prim == PRIM_CYLINDER)
+						{
+							params.primEnabled[qRansacSD::RPT_CYLINDER] = true;
+							primCount++;
+						}
+						else if (prim == PRIM_CONE)
+						{
+							params.primEnabled[qRansacSD::RPT_CONE] = true;
+							primCount++;
+						}
+						else if (prim == PRIM_TORUS)
+						{
+							params.primEnabled[qRansacSD::RPT_TORUS] = true;
+							primCount++;
+						}
+						if (cmd.arguments().empty())
+						{
+							break;
+						}
+						prim = cmd.arguments().takeFirst().toUpper();
+					}
+					if (primCount > 0)
+					{
+						if (!primitiveNames.contains(prim))
+						{
+							cmd.arguments().push_front(prim);
+						}
+					}
+					else
+					{
+						return cmd.error(QObject::tr("No valid parameter: primitive type after \"-%1 %2\"").arg(COMMAND_RANSAC, ENABLE_PRIMITIVE));
+					}
+				}
+				if (cmd.arguments().empty())
+				{
+					break;
+				}
+				param = cmd.arguments().takeFirst().toUpper();
+			}
+			if (!paramNames.contains(param))
+			{
+				cmd.arguments().push_front(param);
+			}
+		}
+		unsigned char primCount = 0;
+		for (unsigned char k = 0; k < 5; ++k)
+		{
+			primCount += (unsigned)params.primEnabled[k];
+		}
+		if (primCount == 0)
+		{
+			cmd.print(QObject::tr("\tDefault Shape Search == %1").arg(PRIM_PLANE));
+			params.primEnabled[qRansacSD::RPT_PLANE] = true;
+		}
+		if (!outputIndividualClouds && !outputIndividualPrimitives && !outputGrouped && !outputIndividualPairs)
+		{
+			cmd.print(QObject::tr("\tDefault output == %1").arg(OUTPUT_GROUPED));
+			outputGrouped = true;
+		}
+
+		for (CLCloudDesc clCloud : cmd.clouds())
```

Comment:
const CLCloudDesc& ?

Reply:
The cloud inside clCloud gets possibly modified when passed to executeRANSAC (normal can be added), also there is no overload const overload for ccCommandLineParser::exportEntity.

Plus I tried with CLCloudDesc& and experienced crashing...